Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesQRS.swift

```diff
@@ -2277,31 +2277,67 @@ public struct StringLiteralExprSyntax: ExprSyntaxProtocol, SyntaxHashable, _Leaf
 
 // MARK: - StringSegmentSyntax
 
-/// A literal segment inside a string segment.
-/// 
-/// - SeeAlso: ``ExpressionSegmentSyntax``
+/// A string literal segment that represents plain text content within a string literal expression.
+///
+/// Used to construct string literals programmatically, especially in macros that need to generate or manipulate string content. For example, in a string literal `"Hello \(name)"`, the `"Hello "` part would be represented by a `StringSegmentSyntax`.
```

Comment:
I wouldn’t say `Used to construct` in the type’s definition because you aren’t necessarily creating syntax trees, you might also just be walking existing trees.